examples/fixed_axis_tsne.py

```diff
@@ -0,0 +1,346 @@
+                if betamax == np.inf or betamax == -np.inf:
+                    beta[i] = beta[i] * 2
+                else:
+                    beta[i] = (beta[i] + betamax) / 2
+            else:
+                betamax = beta[i].copy()
+                if betamin == np.inf or betamin == -np.inf:
+                    beta[i] = beta[i] / 2
+                else:
+                    beta[i] = (beta[i] + betamin) / 2
+
+            # update the value for perb and prob
+            perp, this_prob = cal_perplexity(dist[i], i, beta[i])
+            perp_diff = perp - base_perp
+            tries = tries + 1
+        # record the value for prob
+        pair_prob[i,] = this_prob
+    print("Mean value of sigma: ", np.mean(np.sqrt(1 / beta)))
+
+    return pair_prob
+
+
+def create_triangular_distance_matrix(
+        data):
+    distance_matrix = np.zeros(int((data.shape[0] + 1) * data.shape[0] / 2))
+    size = len(data)
+
+    k = 0
+    for i in range(size):
+        for j in range(i, size):
+            distance_matrix[k] = euclidean(data[i], data[j])
+            k = k + 1
+
+    return distance_matrix
+
+
+def euclidean(x, y):
+    """Standard euclidean distance.
+
+    ..math::
+        D(x, y) = \sqrt{\sum_i (x_i - y_i)^2}
+    """
+    result = 0.0
+    for i in range(x.shape[0]):
+        result += (x[i] - y[i]) ** 2
+    return np.sqrt(result)
+
+
+def kruskal_stress(distance_matrix, projection):
+    size = len(projection)
+    total = len(distance_matrix)
+
+    den = 0
+    num = 0
+
+    for i in numba.prange(size):
+        for j in numba.prange(size):
+            dr2 = euclidean(projection[i], projection[j])
+
+            r = (i + j - math.fabs(i - j)) / 2  # min(i,j)
+            s = (i + j + math.fabs(i - j)) / 2  # max(i,j)
+            drn = distance_matrix[int(total - ((size - r) * (size - r + 1) / 2) + (s - r))]
+
+            num += (drn - dr2) * (drn - dr2)
+            den += drn * drn
+
+    return math.sqrt(num / den)
+
+
+def scale_dataset(data, feature_range):
+    """
+    Helper function for scaling/normalizing
+    """
+    scaler = MinMaxScaler(feature_range=feature_range)
+    data = scaler.fit_transform(data)
+    """
+    data_scalared = data
+    scaler = StandardScaler()
+    data = scaler.fit_transform(data, data_scalared)
+    """
+    return data
+
+
+def preprocess_data(data,
+                    drop_columns_from_dataset=None):
+    """
+    Helper function for preprocessing the data
+    :param data:
+    :param fixed_axis:
+    :param drop_columns_from_dataset:
+    :return: ndarray of shape (n_samples, 2)
+        Starting configuration of the projection result. By default it is ignored,
+        and the starting projection is randomized using starting_projection_mode and random_state.
+        If specified, this must match n_samples.
+    """
+
+    X = data
+
+    if drop_columns_from_dataset is not None:
+        X = np.delete(data, drop_columns_from_dataset, axis=1)
+
+    return X
```

Comment:
these are the same as elsewhere in the code, right? I imagine we could use them instead of having duplicated code

Reply:
Same as above. this is just meant to be an example of fixed axis tsne, we can definitely combine it into the fixed axis force scheme package if that's preferred.